test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/initialgainorloss/InitialGainOrLossControllerSpec.scala

```diff
@@ -0,0 +1,377 @@
+    with ReturnsServiceSupport
+    with ScalaCheckDrivenPropertyChecks
+    with RedirectToStartBehaviour {
+
+  def redirectToStartBehaviour(performAction: () => Future[Result]) =
+    redirectToStartWhenInvalidJourney(
+      performAction, {
+        case _: FillingOutReturn => true
+        case _                   => false
+      }
+    )
+
+  implicit lazy val messagesApi: MessagesApi = controller.messagesApi
+
+  override val overrideBindings = List[GuiceableModule](
+    bind[AuthConnector].toInstance(mockAuthConnector),
+    bind[SessionStore].toInstance(mockSessionStore),
+    bind[ReturnsService].toInstance(mockReturnsService)
+  )
+
+  lazy val controller                  = instanceOf[InitialGainOrLossController]
+  implicit lazy val messages: Messages = MessagesImpl(Lang("en"), messagesApi)
+  val initialGainOrLosses              = sample[AmountInPence].copy(value = 2)
+
+  def sessionWithState(
+    initialGainOrLoss: AmountInPence,
+    disposalDate: DisposalDate
+  ): (SessionData, FillingOutReturn) = {
+    val journey = sample[FillingOutReturn].copy(
+      draftReturn = sample[DraftReturn].copy(
+        initialGainOrLoss = Some(initialGainOrLoss)
+      )
+    )
+
+    SessionData.empty.copy(journeyStatus = Some(journey)) -> journey
+  }
+
+  val allQuestionsAnswered: AmountInPence = initialGainOrLosses
+
+  val (session, journey) = sessionWithState(allQuestionsAnswered, sample[DisposalDate])
+  val updatedDraftReturn =
+    journey.draftReturn.copy(initialGainOrLoss = Some(AmountInPence(0)))
+  val updatedSession         = session.copy(journeyStatus = Some(journey.copy(draftReturn = updatedDraftReturn)))
+  val fillingOutReturnSample = sample[FillingOutReturn]
+
+  "Check Your Answers" when {
+
+    "displaying the page" must {
+      def performAction(): Future[Result] = controller.checkYourAnswers()(FakeRequest())
+
+      behave like redirectToStartBehaviour(() => performAction())
+      "have proper contents" when {
+
+        "loading" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              SessionData.empty.copy(
+                journeyStatus = Some(fillingOutReturnSample)
+              )
+            )
+          }
+
+          checkPageIsDisplayed(
+            performAction(),
+            messageFromMessageKey(
+              "initialGainOrLoss.cya.title"
+            ), { doc =>
+              doc.select("body").html() should include(Messages("initialGainOrLoss.cya.title"))
+              doc.select("#content > article > form").attr("action") shouldBe routes.InitialGainOrLossController
+                .checkYourAnswersSubmit()
+                .url
+            },
+            OK
+          )
+        }
+      }
+    }
+
+    "submitting the page" must {
+      def performAction(): Future[Result] = controller.checkYourAnswersSubmit()(FakeRequest())
+      behave like redirectToStartBehaviour(performAction)
+      "redirect to taskList" when {
```

Comment:
you don't need an inner `in` here - this `when` here can become an `in`

Reply:
Don't know, how I forgot it there, but thanks, Done.